inc/functions-shortcodes.php

```diff
@@ -215,3 +218,43 @@ function members_login_form_shortcode() {
 
 	return wp_login_form( array( 'echo' => false ) );
 }
+
+/**
+ * Filters the login redirect URL to send failed logins back to the
+ * referrer with a query arg of `login=failed`.
+ *
+ * @since 3.2.18
+ *
+ * @param string $redirect_to    The redirect destination URL.
+ * @param string $request        The request URL.
+ * @param object $user           The user object.
+ * @return string                The redirect URL.
+ */
+function members_login_redirect( $redirect_to, $request, $user ) {
+    // If there's a valid referrer, and it's not the default log-in screen
+    if ( ! isset( $_POST['members_redirect_to'] ) ) {
+        return $redirect_to;
+    } elseif ( empty( $user ) || is_wp_error( $user ) ) {
```

Comment:
@ThemeGravity as an extra sanity check, add `isset($user->ID) && $user->ID > 0` to be certain we have a valid User record here.

Reply:
@cartpauj  I removed comment that is no longer valid after previous changes.

I don't quite understand what you mean.

The first check returns default `$redirect_to` if it's not members login form. 

The next condition:

```php
<?php
} elseif ( empty( $user ) || is_wp_error( $user ) ) {
```
checks if login fails (error or no user) so there's no `$user->ID`.

The last check keeps successfully logged in user on the same page.

Reply:
Oh, you're right. Sorry. Ignore this.